### PR TITLE
fixed ConfigurationFactory. use the injected IConfiguration instance for the Create method.

### PR DESCRIPTION
**Pull request by qq362220083**
…ration instance.

Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
#312 
#341 
___
### Bug fix
- Class "ConfigAccessor" method "Get<T>" cannot instantiate type T objects with the latest configuration.
- modify Class "ConfigurationFactory". use the injected IConfiguration instance for the Create method.

### CODE REVIEW

**Comment by snakorse** [comment]
What's the scenario that you can not call the ConfigAccesser.Get successfully.
and why the `builder.AddJsonfile()` and `builder.AddEnvironmentVariables()` got deleted. Please describe more details about this pr.

**Comment by qq362220083** [comment]
We used a configuration management tool. #312 
eg. consul or Apollo

But except for the default configuration of skywalking (`builder.AddSkyWalkingDefaultConfig`), other configuration of skywalking cannot be read with the configuration management tool.
eg. 
1. [SkyApm.Diagnostics.AspNetCore/Config/HostingDiagnosticConfig.cs](https://github.com/SkyAPM/SkyAPM-dotnet/blob/master/src/SkyApm.Diagnostics.AspNetCore/Config/HostingDiagnosticConfig.cs)
2. [SkyApm.Diagnostics.HttpClient/Config/HttpClientDiagnosticConfig.cs](https://github.com/SkyAPM/SkyAPM-dotnet/blob/master/src/SkyApm.Diagnostics.HttpClient/Config/HttpClientDiagnosticConfig.cs)
3. ...

The root cause of this problem is that ConfigurationFactory creates a new IConfigurationBuilder instance, and this new instance does not include the IConfigurationSource collection that has been injected in the current system.

Use `builder.AddConfiguration(_configuration)` can add an existing configuration instance to the IConfigurationBuilder instance as a confiuration source.

The default json files and environment variables is already included by the "_configuration" object.

**Comment by snakorse** [comment]
okay, I understand you, I agree with the adding of `builder.AddConfiguration(_configuration)`, but I think we should not assume that the `_configuration` already contains the default json files or environment variables, so I suggest to keep the `AddJsonfile` and `AddEnvironmentVariables` methonds, we just need make sure the `builder.AddConfiguration(_configuration)` is got called at last.

**Comment by qq362220083** [comment]
Oh, I think we should trust the currently injected `_configuration` object, and only need to attach skywalking customized configuration to the 'builder' object.

**Comment by snakorse** [comment]
> Oh, I think we should trust the currently injected `_configuration` object, and only need to attach skywalking customized configuration to the `builder` object.

I can't agree with that, not skyapm add addtional configuration to `_configuration`, but should be the `_configuration` add addtinal configuration to skyapm, we should keep our skyapm-config work correctly alone without the requirement to the injected IConfiguration object. 
For normal aspnet core webapi project, it indeed can work, but notice that, skyapm is a library, it can used in other environments such as console app with .net fullframework, user injected IConfiguration may have chance lose some configuration method such like the environment variables. In fact, our support for AspNet on fullframework currently even not give user the chance to inject the IConfiguration object, the deletion of these code will stop the user from using environment variables.
So in short, I still recommand keep the json file and environment variables config code, for self-governace.

**Comment by qq362220083** [comment]
:ok_hand: I understand your thoughts, as you wish. 

But i will keep my thoughts :wink:

File: src/SkyApm.Utilities.Configuration/ConfigurationFactory.cs [hunk 1]
```diff
@@ -67,6 +67,8 @@ public IConfiguration Create()
 
             builder.AddEnvironmentVariables();
 
+            builder.AddConfiguration(_configuration);
```
Comment (snakorse) [inline src/SkyApm.Utilities.Configuration/ConfigurationFactory.cs:70]: should perform null check before call the AddConfiguration() method, becase for the AspNet agent, the passed in _configuration is null.

**Review by snakorse — CHANGES_REQUESTED** [review]

File: src/SkyApm.Utilities.Configuration/ConfigurationFactory.cs (hunk 1 above)
Comment (qq362220083) [reply src/SkyApm.Utilities.Configuration/ConfigurationFactory.cs]: Oh, i feel so sorry, i have committed

**Review by snakorse — APPROVED** [review]
LGTM. thanks for your work.
